tests/Utils/TestUtils.mpc

```diff
@@ -1,3 +1,9 @@
-project: dcps_test {
+project: dcps_test, install {
```

Comment:
I don't think we should install anything. We probably shouldn't be installing things in `tests` at all and if we really did, it would probably better belong in `dds` or something else that gets installed. Also `tests/Utils` isn't a path we should install things to as it doesn't follow Unix directory conventions (like those in the [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard), though we don't follow that directly).

I couldn't think of the name at the time, but I was thinking of [`MPC/config/test_root.mpb`](https://github.com/objectcomputing/OpenDDS/blob/master/MPC/config/test_root.mpb) earlier in the meeting. When we go to build tests using an installed OpenDDS, `$TEST_ROOT` can be set to reference things from the source tree. Part of the problem seems to be that the test utils don't have a mpb file where this would be setup centrally. See [`MPC/config/dcps_test_framework.mpb`](https://github.com/objectcomputing/OpenDDS/blob/master/MPC/config/dcps_test_framework.mpb) for an example of this.

Reply:
I'd go further than "I don't think we should install..." -- we can't install these files, at least not without major changes which would seem to be out of scope for this PR.

Many existing tests use StatusMatching.h and build without errors against an installed tree.  What is different in those tests?

If you need to, you can generate makefiles with a `-base` command line argument.

Reply:
In the tick install build the mpc command is `<command name="generate_workspace" options="-type gnuace DDS.mwc -features no_cxx11=0,no_opendds_security=0,no_rapidjson=0 -value_project libout=. -value_project exeout=. -value_project includes+=/tao_builds/jenkins/workspace/dds_doc_ace6tao2_tick_linux_gcc_d1o0/inst"/>`

Reply:
So it appears it's using `-value_project includes+=/tao_builds/jenkins/workspace/dds_doc_ace6tao2_tick_linux_gcc_d1o0/inst"` to force the include on everything. I don't think that's what we should do though, the test utils need a proper mpb.

Reply:
This is by design.  When testing make install it allows the tests to include the headers they need.

Reply:
Yes, I can see that, but it's a brute forcing it when there's an existing finer-grained mechanism that only hands out the include to projects that need it. Brute forcing also could hide issues with using headers that are missing in the install because the headers would always be in the include path. I suggest we make a test utils mpb just for messenger for now, set `$TEST_ROOT` in the CI make install test, and make a task for the future to convert the rest of the test suite and scoreboard to follow suite.

Reply:
The headers that are installed cannot also be found locally since they are removed from the "inst" build tree.

Reply:
There is a separately compiled TestUtils library, but StatusMatching.h (while formally belonging to TestUtils) can be used on its own as header-only.  The separately compiled TestUtils is only used in two tests: Instances and PersistentInfoRepo.